### ISSUE TITLE
Daily Standup Meetings

### ISSUE THREAD

**Issue by GoldHorizon** [FourMen/bulk-club#35]
This is where we'll check in for the stand up meetings each day.  If you have questions/concerns, feel free to voice them here.

**Comment by GoldHorizon** [comment]
# 9/24/2015

**Comment by AnthonyKoutroulis** [comment]
Nice, I'm going to pull my item base class now, I'll probably upload i tomorrow

**Comment by GoldHorizon** [comment]
Yeah @AnthonyKoutroulis I'll make the task for a date class.  Also I started work on the member class, and it's simple enough, but I'm trying to get eclipse to work with the repo.  I'm assuming I just have to install QT to get rid of all this include errors?  Though I find the .cpp files are unable to locate the header files in the /include folder... Will I just have to work on the code in another separate project and copy it into the repo when I'm satisfied?  Not sure.

**Comment by GoldHorizon** [comment]
# 9/25/2015

**Comment by AnthonyKoutroulis** [comment]
@GoldHorizon the include directories are set up for Qt. on eclipse you have to add that folder to your include directories. I can help you set it up if you wanna come by lab on Monday.

**Comment by GoldHorizon** [comment]
@AnthonyKoutroulis I figured out how to include the /include/ folder in the project, but I can't find out how to get rid of the QT include errors still.  Do I need to add more include paths for the QT paths?  I'm not sure where they would be either, trying this all on Linux.

**Comment by AnthonyKoutroulis** [comment]
Then you don't have the Qt libraries, just comment the Qt stuff out for now. We can put it back after we're all setup with Qt.

**Comment by GoldHorizon** [comment]
# 9/26/2015

**Comment by AnthonyKoutroulis** [comment]
Topics of discussion for tonight? I want to squash any environment problems we might have. Maybe, discuss our timeline and rubric .

**Comment by AnthonyKoutroulis** [comment]
BTW. If you are ever waiting on a class that is a dependency for a class you are writing, you can _forward declare_ a dummy dependency class and refactor later. ie: 
```
class RequiredClass; // this is the forward declaration  
class MyClass {  
  private:  
  RequiredClass object;   
}  
```
This will allow you to compile, but obviously it won't do anything until you _include_ and _refactor_ ``` RequiredClass ``` with the actual name of the required class.

**Comment by AnthonyKoutroulis** [comment]
# 9/29/2015

**Comment by AnthonyKoutroulis** [comment]
If you don't want want to do doxygen specific comments for now, that is okay.
That can totally be a 2nd sprint thing I think, we might change things too much to justify the extra work.

**Comment by datowlcs** [comment]
Sorry haven't been on here, I have the InventoryList class done, just have yet to submit it. Currently in the works for branch 10. Should be done soon. Branch 15 is also in the works, but shouldn't be /that/ difficult to add an item to a list (just going to be generic add/delete methods for now).

**Comment by datowlcs** [comment]
@AnthonyKoutroulis I think we should hold off on doxygen till the second sprint. Comment as we go, however let's not preoccupy ourselves with that matter quite yet. Stick to getting the code done. If any code confusion comes up, we can have the code's owner discuss it with the rest of us and comment it in a better fashion.

**Comment by AnthonyKoutroulis** [comment]
@datowlcs I agree, if you look at my last few commits, I included doxygen syntax and it was just extra work. Qt is pretty good at doing most of it, so adding it in the 2nd sprint should be trivial. Let me know if you need help making pull requests. We can do it in lab tomorrow.

**Comment by datowlcs** [comment]
Yeah I was working on push requests today. I still can't get it to work. Have both the previous branches completed... I need to make commits for them separately though as I said. I was meant to post this on 10/1/15 - but It's now 12am that I finally got access to comment.  Anyways, I think I will reinstall my repository (gitshell) and see if I can fix it. I'll keep you guys updated. Otherwise 2 branches of mine are completed and ready for testing.

**Comment by ValkonX11** [comment]
Apologies for the lack of posting. Been having issues getting everything set up on my computer, but I found and resolved what should be the root cause of the problem (needed to install some updates to my OS) and if nothing else goes wrong I should be able to get everything on my end worked out over the rest of the weekend.

**Comment by AnthonyKoutroulis** [comment]
@datowlcs did you configure your git shell username and email, something like ```git config --global.username``` otherwise it wont recognize you. Your account does have push permissions.

**Comment by AnthonyKoutroulis** [comment]
@ValkonX11 No worries, glad to have you up. Please let me know if I can help you get anything fixed, we really should be autonomous by now.

**Comment by datowlcs** [comment]
I uninstalled GitHub, reinstalled, and am currently using PowerShell. As you guys more than likely saw, I was able to pull and create a branch. But I am encountering this error: https://gyazo.com/f7c1f74b3840fadb531c73115cb80ddc

Any suggestions?

**Comment by AnthonyKoutroulis** [comment]
What did you type in that gave you that error, or what were you trying to do?

**Comment by datowlcs** [comment]
trying to push origin

**Comment by AnthonyKoutroulis** [comment]
Did you set up your username and email? Do they match the one here?

**Comment by datowlcs** [comment]
I fixed it. GitShell isn't the easiest format to work in i suppose. I switch to GitBash and it's much simpler. Another question though, looking in master, would you prefer I made the inventory list (adding an item method) a separate function for now until we bring everything together?

**Comment by AnthonyKoutroulis** [comment]
Just make the inventory list class declarations and definitions.  I have the item class waiting for review, so you can look at that to know what to include and use (also, comment on there if you think something needs to be changed/added).  If you want to make sure your class compiles you can *forward declare* a fake item class, see my comment on this thread on 9/28

**Comment by AnthonyKoutroulis** [comment]
Also, gitbash is way better.  All of the graphical git stuff kinda just complicates things.  Terminal git just works

**Comment by GoldHorizon** [comment]
Hey sorry, I know I haven't been on here too much.  How is everyone doing on their stories/parts?  How far are we along?  I won't have time to code before class tomorrow so I'm going to get everything done for my part by tonight if all goes well.  Are we going to have to merge everything and create a decent gui for tomorrow?  Will that be done tonight or tomorrow before class?

**Comment by GoldHorizon** [comment]
@AnthonyKoutroulis I don't know how far everyone is, or if we've all started, but I'm wondering what we will have to show tomorrow.  I should be able to get this member list class done and allow creating and deleting members (though I will probably end up breaking agile flow), so if I can do that can we try to get a small GUI ready? I don't know how much time you'll have tomorrow.

**Comment by datowlcs** [comment]
@GoldHorizon I'd recommend just having 2 branches done each. Currently working out some kinks. Mine should be ready around noon tmw and uploaded to testing. If we have any trouble, I'd suggest we bring any modified code we have separately on a usb to class and merge it ask the professor goes through everyone's projects.

**Comment by ValkonX11** [comment]
Well, despite my best efforts, Eclipse refuses to build anything on my home computer (and I don't have QT set up on it). Gives me a makefile error that I can't seem to find the cause for. On the other hand, I have what I think should be functional code (i.e. can add and remove members from the list) for MemberList using the vector functions written, though, so I can bring that up on the computers in the lab tomorrow, make sure it works, etc. there.

**Comment by GoldHorizon** [comment]
@datowlcs Okay good to know, hopefully we can merge everything

@ValkonX11 I was working on the member list last night haha, but you have probably done more then me, have you been doing in eclipse? Also have you tried running qmake in QT to fix the build errors?

**Comment by GoldHorizon** [comment]
@ValkonX11 Nevermind I understand you dont have QT set up. Can you push your work to git so we can all look at it as soon as possible?

**Comment by AnthonyKoutroulis** [comment]
We should be able to visually inspect all your errors on review. Everyone just submit your code, and we will review. Also the testing branch is designed to handle all build testing.

**Comment by AnthonyKoutroulis** [comment]
However, everyone _needs_ to have Qt set up, and you _should_ be using it. Qt is a **requirement** for this project.

**Comment by AnthonyKoutroulis** [comment]
@GoldHorizon Can you make and assign issues to create test clients?

**Comment by GoldHorizon** [comment]
@AnthonyKoutroulis Yeah, should I do that for each person? Or each story?

**Comment by AnthonyKoutroulis** [comment]
Hmm... Well maybe we can just wait and rework the stories together on Friday. That way we can figure out what needs to be tested with what story. I'm sure we can reorganize the work done with tests into complete stories and get credit on Monday.

**Comment by datowlcs** [comment]
Regarding test clients, would we all be making our own test clients for our individual methods? I have kind of reworked some of mine to fit a basic console interface. I'd personally suggest we do this together. (this post was meant for 10/6/15)

**Comment by AnthonyKoutroulis** [comment]
@datowlcs Yup I agree. we can probably right some sort of unified console client together.  We could also probably just make a new ui for each test too.

**Comment by GoldHorizon** [comment]
@datowlcs @AnthonyKoutroulis We could separate menu functions for each test/story, then just write a small menu that has the user choose between which menu to go to. Would be pretty nice and modular I think

**Comment by datowlcs** [comment]
@GoldHorizon That also works. I have mine done for inventory (adding and deleting) items. There wasn't much to test or really implement there.

**Comment by datowlcs** [comment]
Anyhow, are we still working/meeting up at 1:00/1:30pm tomorrow? I don't really have access to a laptop, which is my only drawback.

**Comment by GoldHorizon** [comment]
@datowlcs We should definitely try to meet tomorrow, I'm available pretty much any time in the afternoon.  We may decide to meet at one of our houses, and hopefully that could get you access to another computer.  We'll see what @AnthonyKoutroulis thinks on where we can go.

**Comment by GoldHorizon** [comment]
If it would be easier for everyone tomorrow (saturday) would work for meeting up as well, I would be available after 2 or so

**Comment by AnthonyKoutroulis** [comment]
I'm going to be at JPL on Saturday.

**Comment by GoldHorizon** [comment]
Alright so today would be best. Where/when can we meet?

**Comment by datowlcs** [comment]
I'm free, have been free as well. If not we can work at home separately and communicate via Skype or just text based,

**Comment by GoldHorizon** [comment]
@datowlcs Yeah we could Skype, you said you don't actually have access to a laptop?  If we did meet then I'm not sure what you could do without a computer.  So it's possible Skype is our best bet.

**Comment by GoldHorizon** [comment]
I'm going to start work on a console menu for one of the stories tonight.  We can probably finish them all tomorrow in/before lab and put them together with one overarching menu.  For now I'm assuming it will be easiest to use the testing branch as a base for new ones, since that contains all of our previous code.

**Comment by AnthonyKoutroulis** [comment]
Cool. Can everyone meet in lab before class tomorrow?

**Comment by datowlcs** [comment]
My free time is roughly the same time we met before. Around 4:45/5:00pm.

**Comment by GoldHorizon** [comment]
Same here, so I will be able to come in 10 or 15 minutes before class starts

**Comment by datowlcs** [comment]
I just barely got home and class is still about a 25 minute drive for me. Not sure if I can make it in time. I can submit my testing to you guys quickly, but I might barely just make it to lecture.

**Comment by datowlcs** [comment]
Sorry about missing lab today guys. I just barely got home and had to deal with family matters. I might be able to slip into lab around like 8:20 if you guys are still present and have yet to go through QT. Otherwise I have looked into it and will look into it more before friday.

**Comment by GoldHorizon** [comment]
It's all good, we're still on for meeting tomorrow around 2 at @AnthonyKoutroulis' house.  We just need the address and then we'll be set, so if it's not posted here by tomorrow afternoon we can just shoot him a text.
